lib/module-resolvers/file-resolver.js

```diff
@@ -38,36 +40,75 @@ class FileResolver {
         }
     }
 
-    _search(module_path) {
+    _search(module_path, module_name, importer_path) {
         var self = this;
 
+        logger.debug(`Resolving module_path=${module_path} module_name=${module_name} importer_path=${importer_path}`);
+
         var k, paths = self._search_paths;
         paths = paths.concat(self._additional_search_paths);
         if (path.extname(module_path) !== '.juttle') {
             module_path = module_path + '.juttle';
         }
-        for (k = 0; k < paths.length; ++k) {
-            var filename = paths[k] + '/' + module_path;
-            if (self._is_file(filename)) {
-                try {
-                    return fs.readFileSync(filename, 'utf8');
-                }
-                catch (e) {
-                    // continue looking for a module we can successfully load
+
+        if (module_path.indexOf('./') === 0 ||
+            module_path.indexOf('../') === 0) {
+
+            // module_path is a relative path and resolve it relative
+            // to the base directory of the file doing the import, or
+            // the current working directory.
+            let dir = process.cwd();
+            if (path.isAbsolute(importer_path)) {
+                dir = path.dirname(importer_path);
+            }
+            let filename = path.resolve(dir, module_path);
+            try {
+                return {
+                    source: fs.readFileSync(filename, 'utf8'),
+                    filename: filename
+                };
+            }
+            catch (e) {
+                return false;
+            }
+        } else if (path.isAbsolute(module_path)) {
+            // Simply try to read the file
+            try {
+                return {
+                    source: fs.readFileSync(module_path, 'utf8'),
+                    filename: module_path
+                };
+            }
+            catch (e) {
+                return false;
+            }
+        } else {
+            for (k = 0; k < paths.length; ++k) {
+                var filename = paths[k] + '/' + module_path;
+                if (self._is_file(filename)) {
+                    try {
+                        return {
+                            source: fs.readFileSync(filename, 'utf8'),
+                            filename: filename
+                        };
+                    }
+                    catch (e) {
+                        // continue looking for a module we can successfully load
+                    }
```

Comment:
It would be nice to split the relative module lookup and system module lookup into separate branches.